src/dia/Cell.mjs

```diff
@@ -298,6 +300,8 @@ export const Cell = Backbone.Model.extend({
 
                 z -= cells.length;
 
+                this._sortCellsByZIndex(cells); // we must preserve stacking
```

Comment:
There is a utility available for array sorting.
```
const sortedCells = util.sortBy(cells, cell => cell.get('z') || 0);
```
 *(Note: it does not sort array in-place)*

Reply:
Perfect, I'll use it. 👍